src/jvmTest/kotlin/test.kt

```diff
@@ -140,29 +142,31 @@ internal class DiamondTests {
                 println(assumptions)
 
                 val timeKoSat = (measureTimeMillis {
-                    solution = solveCnf(readCnfRequests(input).first())
+                    solution = solveWithAssumptions(readCnfRequests(input).first())
                 }.toDouble() / 1000).toString()
 
                 val timeMiniSat = measureTimeMillis { isSolution = processMiniSatSolver(input) }.toDouble() / 1000
 
                 val checkRes = if (checkKoSatSolution(solution, input, isSolution)) "OK" else "WA"
 
                 println(
-                    buildPadding(listOf(
-                        filename.dropLast(format.length), // test name
-                        timeKoSat,
-                        timeMiniSat.toString(),
-                        checkRes,
-                        if (isSolution) "SAT" else "UNSAT"
-                    ))
+                    buildPadding(
+                        listOf(
+                            filename.dropLast(format.length), // test name
+                            timeKoSat,
+                            timeMiniSat.toString(),
+                            checkRes,
+                            if (isSolution) "SAT" else "UNSAT"
+                        )
+                    )
                 )
             }
         }
     }
 
     @Test
     fun test() {
-        assertEquals(runTests("src/jvmTest/resources/"), true)
+        runTests("src/jvmTest/resources/")
```

Comment:
Why do you put your data inside "resources" dir if you are not using it as an actual "resource" ([in Java terms](https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getResource(java.lang.String)))? You are just hardcoding the path (and later even transform it into an absolute one, which makes no sense for resources, but is perfectly valid for "just files", as in your case), so you could equally-happily place all CNFs into `/data/` folder in root without cluttering the jar bundle.
(Actually, there is no big problem, since you are inside `test` configuration, not `main`, but anyway, looks like not a good practice (or maybe I'm too old for this))